test/e2e/workflow/workflow.test.ts

```diff
@@ -0,0 +1,380 @@
+import { Task } from "kaibocai-durabletask-js/task/task";
+
+
+describe("Workflow", () => {
+    const grpcEndpoint = "localhost:4001";
+    let workflowClient: WorkflowClient;
+    let workflowRuntime: WorkflowRuntime;
+
+    beforeAll(async () => { });
+
+    beforeEach(async () => {
+        // Start a worker, which will connect to the sidecar in a background thread
+        workflowClient = new WorkflowClient(grpcEndpoint);
+        workflowRuntime = new WorkflowRuntime(grpcEndpoint);
+    });
+
+    afterEach(async () => {
+        await workflowRuntime.stop();
+        await workflowClient.stop();
+    });
+
+    it("should be able to run an empty orchestration", async () => {
+        let invoked = false;
+        const emptyWorkflow: TWorkflow = async (_: WorkflowContext, __: any) => {
+            invoked = true;
+        }
+        workflowRuntime.registerWorkflow(emptyWorkflow);
+        await workflowRuntime.start();
+
+        const id = await workflowClient.scheduleNewWorkflow(emptyWorkflow);
+        const state = await workflowClient.waitForWorkflowCompletion(id, undefined, 30);
+
+        expect(invoked);
+        expect(state);
+        expect(state?.name).toEqual(getFunctionName(emptyWorkflow));
+        expect(state?.instanceId).toEqual(id);
+        expect(state?.workflowFailureDetails).toBeUndefined();
+        expect(state?.runtimeStatus).toEqual(WorkflowRuntimeStatus.COMPLETED);
+    });
+
+    it("should be able to run an activity sequence", async () => {
+        const plusOne = async (_: WorkflowActivityContext, input: number) => {
+            return input + 1;
+        };
+
+        const sequence: TWorkflow = async function* (ctx: WorkflowContext, startVal: number): any {
+            const numbers = [startVal];
+            let current = startVal;
+
+            for (let i = 0; i < 10; i++) {
+                current = yield ctx.callActivity(plusOne, current);
+                numbers.push(current);
+            }
+
+            return numbers;
+        };
+
+        workflowRuntime.registerWorkflow(sequence).registerActivity(plusOne);
+        await workflowRuntime.start();
+
+        const id = await workflowClient.scheduleNewWorkflow(sequence, 1);
+        const state = await workflowClient.waitForWorkflowCompletion(id, undefined, 30);
+
+        expect(state);
+        expect(state?.name).toEqual(getFunctionName(sequence));
+        expect(state?.instanceId).toEqual(id);
+        expect(state?.workflowFailureDetails).toBeUndefined();
+        expect(state?.runtimeStatus).toEqual(WorkflowRuntimeStatus.COMPLETED);
+        expect(state?.serializedInput).toEqual(JSON.stringify(1));
+        expect(state?.serializedOutput).toEqual(JSON.stringify([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]));
+    }, 31000);
+
+    it("should be able to run fan-out/fan-in", async () => {
+        let activityCounter = 0;
+
+        const increment = (ctx: WorkflowActivityContext, _: any) => {
```

Comment:
Let's rename activities to `xyzActivity` (e.g., `incrementActivity`) and workflows to `xyzWorkflow` (e.g., `sequenceWorkflow`).